web/src/main/java/ca/hendriks/bartender/web/ingredient/IngredientsRestController.java

```diff
@@ -28,4 +29,18 @@ public Ingredient addIngredient(@RequestBody Ingredient ingredient) {
         return ingredientRepository.save(ingredient);
     }
 
+    @PutMapping
+    public void updateIngredient(@RequestBody List<String> ingredientChangeList){
+        final int ingredientToBeUpdatedId = ingredientRepository.findByName(ingredientChangeList.get(0)).getId();
+        final IngredientType updatedIngredientType = IngredientType.valueOf(ingredientChangeList.get(3));
+        final Ingredient updatedIngredient = new Ingredient(ingredientToBeUpdatedId, ingredientChangeList.get(2), updatedIngredientType);
+        ingredientRepository.save(updatedIngredient);
+    }
+
+    @DeleteMapping
+    public void deleteIngredient(@RequestBody Object ingredientName){
```

Comment:
`String` instead of `Object` ?

Reply:
So.. yes originally. But the test simply wasn't passing. It turns out that the "Vodka" in the test was being sent (through json) as ""Vodka"" and wasn't being found in the repo.

This suggested to pass an object instead of a string, and it works
https://stackoverflow.com/questions/64990357/parsing-a-request-body-containing-a-quoted-string-as-json-in-spring-boot-2#:~:text=let%20Spring%20handle%20RequestBody%20as%20an%20Object 

I could pass a whole ingredient, but that seems a bit unnecessary (especially since name acts as a PK)

Reply:
> (especially since name acts as a PK)

should name act as PK? What if your name has spaces, like "Peach Schnapps". Then your URI would look like

`/ingredients/Peach Schnapps`

uh-oh... which isn't very copy-pastable, and most browsers will convert it to `/ingredients/Peach%20Schnapps` which can be problematic. And what if Peach Schnapps get's renamed to McGillicudy's Peach Schnapps... oh man, every try to rename a record's primary key in a database?

That's why a common practice is to use a _surrogate key_, such as a database sequence or a UUID. They are unique, never need to be renamed, and have no special characters. Better, ya?

Reply:
btw 

> It turns out that the "Vodka" in the test was being sent (through json) as ""Vodka"" and wasn't being found in the rep

sounds like the problem here was that your input is wrong. The name isn't 

`"Vodka"`

it's 

`Vodka`

Reply:
I don't understand why it gave that error. I never changed the feature file, even from what it is now. The only change is that it sent it via Object vs String
    When the administrator deletes "Vodka"
    
It's the same format as the add, even:
    When the administrator adds "Vodka" into category "SPIRITS",

Reply:
> > (especially since name acts as a PK)
> 
> should name act as PK? What if your name has spaces, like "Peach Schnapps". Then your URI would look like
> 
> `/ingredients/Peach Schnapps`
> 
> uh-oh... which isn't very copy-pastable, and most browsers will convert it to `/ingredients/Peach%20Schnapps` which can be problematic. And what if Peach Schnapps get's renamed to McGillicudy's Peach Schnapps... oh man, every try to rename a record's primary key in a database?
> 

My bad, I thought we removed the Ingredient's ID in commit f0d565bac07e390e00d4b10f3d029185df73cee1. It's just hibernate's responsibility now